modules/web-console/frontend/app/modules/agent/AgentManager.service.ts

```diff
@@ -104,15 +119,23 @@ class ConnectionState {
 
         if (_.isEmpty(this.clusters))
             this.cluster = null;
-        else if (_.isNil(this.cluster))
-            this.cluster = _.head(clusters);
+        else if (_.isNil(this.cluster)) {
+            const restoredCluster = _.find(clusters, {id: localStorage.clusterId});
```

Comment:
access to localStorage should be wrapped by try catch block, because it throw exception  in safari incognito mode

Reply:
Wrapped by try catch block.